src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheckTest.java

```diff
@@ -795,6 +795,27 @@ public void testTrimOption() throws Exception {
                 expected);
     }
 
+    @Test
+    public void testUnmodifiableProperties() {
+        final ImportOrderCheck checkObj = new ImportOrderCheck();
+
+        final String[] expectedGroups = {"groups1", "groups2"};
+        final String[] expectedStaticGroups = {"static1", "static2"};
+        checkObj.setGroups(expectedGroups);
+        checkObj.setStaticGroups(expectedStaticGroups);
+
+        final String[] groups = TestUtil.getInternalState(checkObj, "groups");
+        final String[] staticGroups = TestUtil.getInternalState(checkObj, "staticGroups");
+        groups[0] = "modified";
+        staticGroups[0] = "modified";
+        assertWithMessage("Property 'groups' is not unmodifiable")
+                .that(expectedGroups[0])
+                .isNotEqualTo("modified");
```

Comment:
if Check going to have not Input based test, it should have comment to explain why it can not be covered by Input based test, it should be be very good reason.
if possible to be Input based, must be Input based.

Reply:
I've tried to make this input-based, but I could not. This test is to kill Pitest. Pitest will automatically remove the function call to `UnmodifiableCollectionUtil.copyOfArray()` and there is no difference after such mutation. The only way this mutation can be detected is through changing the reference (as above), which cannot be done through input files.

Reply:
What this test try to cover? Pitest mutation only?

Reply:
Yes, only for covering Pitest mutation.

@romani I have some issues with Pitest running locally, I need some more time to fix it. But just for your quick reference, this test case is to cover the following two usages of `UnmodifiableCollectionUtil.copyOfArray()` in `ImportOrderCheck.java`.

```java
public void setGroups(String... packageGroups) {
        groups = UnmodifiableCollectionUtil.copyOfArray(packageGroups, packageGroups.length);
        groupsReg = compilePatterns(packageGroups);
}
```

```java
public void setStaticGroups(String... packageGroups) {
        staticGroups = UnmodifiableCollectionUtil.copyOfArray(packageGroups, packageGroups.length);
        staticGroupsReg = compilePatterns(packageGroups);
}
```

As mentioned in https://github.com/checkstyle/checkstyle/pull/14740#discussion_r1564228923, Pitest will remove the call to `UnmodifiableCollectionUtil.copyOfArray()` and complains no unit test detects this removal.

Reply:
I can not allow this, we did a lot of efforts to kill all pitest mutations by Inputs.
Please share full mutation survival block or point to CI execution when you detected it, to let me see all details.

Reply:
Please add https://github.com/checkstyle/checkstyle/blob/69337c7913430eea2e6618046e50102f770ab68d/pom.xml#L3072
To https://github.com/checkstyle/checkstyle/blob/69337c7913430eea2e6618046e50102f770ab68d/pom.xml#L3635

Reply:
Sure, I've changed `pom.xml` and pushed again. I will check the Pitest result by CI.